protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java

```diff
@@ -0,0 +1,254 @@
+
+@Component
+/**
+ * XmpDetector class
+ * @author rdk <rdk@krupczak.org>
+ * @version $Id: $
+ *
+ */
+@Scope("prototype")
+public class XmpDetector implements SyncServiceDetector
+{
+    // class variables
+
+    private static final String DEFAULT_SERVICE_NAME = "XMP";
+    private static final Logger LOG = LoggerFactory.getLogger(XmpDetector.class);
+    private static int XMP_DEFAULT_TIMEOUT = 3000;
+    private static int XMP_DEFAULT_RETRIES = 1;
+    
+    // instance variables
+    SocketOpts sockopts;
+    XmpConfig protoConfig;
+    int xmpPort;
+    int xmpTimeout;
+    int xmpRetries;
+    String xmpAuthenUser;
+    String xmpServiceName; 
+    String m_ipMatch;
+    Date createTimeDate;
+
+    /* 
+     * @param serviceName a {@link java.lang.String} object
+     * @param port an int specifying tcp port number
+     */
+    
+    public XmpDetector(String serviceName, int port)
+    {
+	//super(serviceName,port,XMP_DEFAULT_TIMEOUT,XMP_DEFAULT_RETRIES);
+	
+        // set default config
+        xmpPort = port;
+        xmpAuthenUser = new String("xmpUser"); 
+        xmpTimeout = XMP_DEFAULT_TIMEOUT; /* millseconds */
+        xmpRetries = XMP_DEFAULT_RETRIES;
+
+        // get socket opts
+        sockopts = new SocketOpts();
+
+        xmpServiceName = new String(serviceName);
+
+	// very important to set to null, not 0-len string
+	// as provisiond's ip range matching functionality
+	// will not correctly match if 0-len instead of null
+	
+	//m_ipMatch = new String("");
+	m_ipMatch = null;  
+
+	createTimeDate = new Date();
+
+	if (LOG == null) {
+	   System.out.println("XmpDetector created, but null LOG");
+	}
+	else {
+	    LOG.debug("XmpDetector created, service "+xmpServiceName+" at "+createTimeDate);
+	}
+
+    } /* XmpDetector */
+
+    public XmpDetector()
+    {
+	this(DEFAULT_SERVICE_NAME,Xmp.XMP_PORT);
+        
+    } /* XmpDetector */
+
+    public String getServiceName() {
+	LOG.debug("XmpDetector: getServiceName");
+	return xmpServiceName;
+    }
+  
+    public void setServiceName(String newServiceName) 
+    {
+	LOG.debug("XmpDetector: setServiceName to "+newServiceName);
+    	xmpServiceName = new String(newServiceName);         
+    }
+
+    public void init() { onInit(); }
+
+    public void onInit()
+    {
+        LOG.debug("XmpDetector: onInit starting");
+	
+        // try to get configuration
+        try { 
+            XmpConfig protoConfig;
+
+	    XmpConfigFactory.init();
+            protoConfig = XmpConfigFactory.getInstance().getXmpConfig();
+            if (protoConfig.hasPort())
+               xmpPort = protoConfig.getPort();
+            if (protoConfig.hasTimeout())
+               xmpTimeout = protoConfig.getTimeout();
+            if (protoConfig.getAuthenUser() != null)
+               xmpAuthenUser = protoConfig.getAuthenUser();
+
+            sockopts.setConnectTimeout(xmpTimeout);	    
+
+        } catch (Throwable e) {
+            if (LOG != null)
```

Comment:
Its necessary because I'm setting up SSL socket stuff and I have to catch certificate issues.